mdtk/pytorch_datasets.py

```diff
@@ -1 +1,327 @@
+        value = int(cmd_str[1:])
+        if cmd == 'o':
+            note_on_pitch += [value]
+            note_on_time += [curr_time]
+        elif cmd == 'f':
+            note_off_pitch += [value]
+            note_off_time += [curr_time]
+        elif cmd == 't':
+            curr_time += value
+        else:
+            raise ValueError(f'Invalid command {cmd}')
+    df = pd.DataFrame(columns=['onset', 'track', 'pitch', 'dur'], dtype=int)
+    for ii, (pitch, onset) in enumerate(zip(note_on_pitch, note_on_time)):
+        note_off_idx = note_off_pitch.index(pitch)  # gets first instance
+        note_off_pitch.pop(note_off_idx)
+        off = note_off_time.pop(note_off_idx)
+        dur = off - onset
+        track = 0
+        df.loc[ii] = [onset, track, pitch, dur]
+    
+    return df
+    
+
+
+def transform_to_torchtensor(output):
+    return {key: torch.tensor(value) for key, value in output.items()}
+
+
+
+# This is adapted from https://github.com/codertimo/BERT-pytorch/blob/master/bert_pytorch/dataset/dataset.py
+class CommandDataset(Dataset):
+    def __init__(self, corpus_path, vocab, seq_len, encoding="utf-8",
+                 corpus_lines=None, in_memory=True, transform=None):
+        """
+        Returns data required for ACME tasks. The returned item
+    
+        Parameters
+        ----------
+        corpus_path : str
+            Path to document containing the corpus of data. Each line is comma
+            separated and contains the degraded command string, clean command
+            string, then the degadation id label (0 is no degradation).
+            
+        vocab : Vocab class
+            A Vocab class object (see CommandVocab above). This is used to
+            convert the string commands to integers and serves as an easy
+            way of getting them back again.
+        
+        seq_len : int
+            The maximum length for a sequence (all sequences will be padded
+            to this length)
+        
+        encoding : str
+            Encoding to use when opening the corpus file.
+        
+        corpus_lines : int
+            Optional, if you know the number of lines in the corpus, supplying
+            the number here saves counting the lines in the file.
+        
+        in_memory : bool
+            Whether to store data in memory, or read from disk. N.B. If reading
+            from disk, the __get_item__ method ignores the item index and just
+            reads the next line of the file. This means batching will not be
+            random when used with a dataloader.
+        
+        transform: func
+            The output from __get_item__ is a dictionary of numpy arrays.
+            The function transform is applied to the dictionary before it is
+            returned so, for example, it can be used to convert all data to
+            torch tensors.
+    
+        Returns
+        -------
+        df : pd.DataFrame
+            The pandas DataFrame representing the note data
```

Comment:
I don't think it returns anything.

Reply:
Haha, yes, sorry I copied a docstring above and forgot to remove this. I'll remove.